paperqa/docs.py

```diff
@@ -475,11 +471,15 @@ def delete(
 
     async def _build_texts_index(self, embedding_model: EmbeddingModel) -> None:
         texts = [t for t in self.texts if t not in self.texts_index]
+        # For any embeddings we are supposed to lazily embed, embed them now
         to_embed = [t.text for t in texts if t.embedding is None]
         if to_embed:
-            embeddings = await embedding_model.embed_documents(texts=to_embed)
-            for t, e in zip(texts, embeddings, strict=True):
-                t.embedding = e
+            for t, t_embedding in zip(
+                texts,
+                await embedding_model.embed_documents(texts=to_embed),
```

Comment:
to_embed and texts will not be same shape. Think this is a bug.

Reply:
Oh nice catch

So you think it should have `zip(to_embed, ..., strict=True)` instead of `zip(texts, ..., strict=True)`?

I will also look into if this code is covered by our tests

Reply:
Only want to embed the to_embed list here.

Reply:
Okay just pushed the fix, thanks for confirming